packages/components/src/components/ColorPicker/ColorPickerInput.tsx

```diff
@@ -23,43 +23,59 @@ type ColorPickerInputProps = {
      * callback for clearing the color
      */
     onClear?: () => void;
+    /**
+     * Event handler called when the color picker input is clicked
+     */
+    onClick?: () => void;
     /**
      * The test id of the color picker input
      */
     'data-test-id'?: string;
 } & CommonAriaAttrs;
 
 export const ColorPickerInput = (
-    { id, currentColor, onClear, 'data-test-id': dataTestId = 'color-picker-input', ...props }: ColorPickerInputProps,
+    {
+        id,
+        currentColor,
+        isOpen,
+        onClear,
+        onClick,
+        'data-test-id': dataTestId = 'color-picker-input',
+        ...props
+    }: ColorPickerInputProps,
     forwardedRef: ForwardedRef<HTMLDivElement>,
 ) => {
     const colorNameId = useId();
 
     return (
         <div id={id} className={styles.root} {...props} ref={forwardedRef} data-test-id={dataTestId}>
-            {currentColor?.red !== undefined ? (
-                <div
-                    aria-describedby={colorNameId}
-                    className={styles.colorIndicator}
-                    style={{ backgroundColor: colorToCss(currentColor) }}
-                />
-            ) : (
-                <>
-                    <IconDroplet size={16} />
-                    <span>Select Color</span>
-                </>
-            )}
+            <button className={styles.button} onClick={onClick} data-color-input-select>
+                {currentColor?.red !== undefined ? (
+                    <div
+                        aria-describedby={colorNameId}
+                        className={styles.colorIndicator}
+                        style={{ backgroundColor: colorToCss(currentColor) }}
+                    />
+                ) : (
+                    <>
+                        <IconDroplet size={16} />
+                        <span>Select Color</span>
+                    </>
+                )}
 
-            <span id={colorNameId} className={styles.colorName}>
-                {currentColor?.name}
-            </span>
-            {onClear && (
-                <button type="button" aria-label="Clear color" onClick={onClear} className={styles.clear}>
-                    <IconTrashBin size={16} />
-                </button>
-            )}
-            <div>
-                <IconCaretDown size={16} className={styles.caret} />
+                <span id={colorNameId} className={styles.colorName}>
```

Comment:
given this will be read by the screen reader, perhaps we should apply `aria-hidden` on the color swatch